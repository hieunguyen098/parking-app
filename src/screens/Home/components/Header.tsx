import { StyleSheet, Text, View, Image, TextInput } from 'react-native';
import React from 'react';
import { LinearGradient } from 'expo-linear-gradient';
import Ionicons from '@expo/vector-icons/Ionicons';
import SearchInput from './SearchInput';

const Header = () => {
    return (
        <LinearGradient colors={['rgba(255,149,58,1)', 'rgba(255,149,58,0)']} style={styles.headerContainer}>
            <View style={styles.firstLineContainer}>
                <Text style={styles.name}>Chào Cao Thanh Bình !</Text>
                <View style={styles.rightHeaderContainer}>
                    <View style={styles.iconContainer}>
                        <Ionicons name="notifications-circle" size={38} color="rgba(0,0,0,0.3)" />
                        <Text style={styles.badge}>3</Text>
                    </View>
                    <Image
                        source={{
                            uri: 'https://media.gq.com/photos/56bcb218cdf2db6945d2ef93/master/pass/bieber-coverstory-square.jpg',
                        }}
                        style={styles.avatar}
                    />
                </View>
            </View>
            <View style={styles.bannerContainer}>
                <Image
                    source={{
                        uri: 'https://cdn.vectorstock.com/i/1000x1000/75/78/parking-conceptual-web-banner-car-park-vector-12827578.webp',
                    }}
                    style={styles.banner}
                />
            </View>
            <SearchInput />
        </LinearGradient>
    );
};

export default Header;

const styles = StyleSheet.create({
    headerContainer: {
        paddingTop: 40,
        paddingBottom: 32,
        paddingHorizontal: 16,
    },
    name: {
        fontSize: 20,
        color: '#FFFFFF',
        fontWeight: '600',
    },
    rightHeaderContainer: {
        flexDirection: 'row',
        alignItems: 'center',
    },
    iconContainer: {
        position: 'relative',
        marginRight: 8,
    },
    badge: {
        position: 'absolute',
        right: 0,
        top: 0,
        backgroundColor: 'red',
        paddingHorizontal: 5,
        borderRadius: 50,
        color: 'white',
        fontSize: 12,
    },
    firstLineContainer: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        alignItems: 'center',
    },
    avatar: {
        width: 32,
        height: 32,
        borderRadius: 16,
    },
    bannerContainer: {
        marginVertical: 10,
    },
    banner: {
        width: '100%',
        height: 100,
        borderRadius: 10,
    },
});
