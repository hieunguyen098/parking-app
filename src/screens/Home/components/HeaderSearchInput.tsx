import { StyleSheet, View, TextInput } from 'react-native';
import Ionicons from '@expo/vector-icons/Ionicons';

import React from 'react';
import { useNavigation } from '@react-navigation/native';
import { GlobalStyles } from '../../../constants/style';
import SearchInput from '../../../components/SearchInput/SearchInput';

const HeaderSearchInput = () => {
    const navigation: any = useNavigation();
    const handleSearch = () => {
        navigation.navigate('SearchResults');
    };

    return <SearchInput placeholder="Tìm kiếm bãi đỗ" onSearch={handleSearch} />;
};

export default HeaderSearchInput;

const styles = StyleSheet.create({
    searchContainer: {
        flex: 1,
        flexDirection: 'row',
        backgroundColor: 'white',
        height: 44,
        marginVertical: 10,
        alignItems: 'center',
        borderRadius: 10,
        paddingHorizontal: 10,
        elevation: 2,
    },
    inputSearch: {
        flex: 1,
        fontSize: 20,
        marginLeft: 10,
    },
});
