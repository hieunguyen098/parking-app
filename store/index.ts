import { configureStore } from '@reduxjs/toolkit';
import { authReducer } from './slices/authSlice';
import { generalReducer } from './slices/generalSlice';

export const store = configureStore({
    reducer: {
        auth: authReducer,
        general: generalReducer,
    },
});

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
// Inferred type: {posts: PostsState, comments: CommentsState, users: UsersState}
export type AppDispatch = typeof store.dispatch;
