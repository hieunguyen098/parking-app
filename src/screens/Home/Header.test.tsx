import React from 'react';
import { render } from '@testing-library/react-native';
import { NavigationContainer } from '@react-navigation/native';
import HeaderSearchInput from './components/HeaderSearchInput';
import Header from './components/Header';
import { Provider } from 'react-redux';
import { store } from '../../../store';
test('test header search input', () => {
    const component = (
        <Provider store={store}>
            <NavigationContainer>
                <Header />
            </NavigationContainer>
        </Provider>
    );

    const tree = render(component);
    expect(tree).toMatchSnapshot();
});
