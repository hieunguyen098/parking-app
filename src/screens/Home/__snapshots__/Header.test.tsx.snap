// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test header search input 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    Object {
      "colors": Array [
        4294939962,
        16749882,
      ],
      "endPoint": undefined,
      "locations": undefined,
      "startPoint": undefined,
    }
  }
  style={
    Object {
      "paddingBottom": 32,
      "paddingHorizontal": 16,
      "paddingTop": 8,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#FFFFFF",
          "fontSize": 20,
          "fontWeight": "600",
        }
      }
    >
      Chào, 
       !
    </Text>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "marginRight": 8,
            "position": "relative",
          }
        }
      >
        <Text />
        <Text
          style={
            Object {
              "backgroundColor": "red",
              "borderRadius": 50,
              "color": "white",
              "fontSize": 12,
              "paddingHorizontal": 5,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          3
        </Text>
      </View>
      <Image
        source={
          Object {
            "uri": "https://media.gq.com/photos/56bcb218cdf2db6945d2ef93/master/pass/bieber-coverstory-square.jpg",
          }
        }
        style={
          Object {
            "borderRadius": 16,
            "height": 32,
            "width": 32,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Object {
        "marginVertical": 10,
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://cdn.vectorstock.com/i/1000x1000/75/78/parking-conceptual-web-banner-car-park-vector-12827578.webp",
        }
      }
      style={
        Object {
          "borderRadius": 10,
          "height": 100,
          "width": "100%",
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "white",
        "borderRadius": 10,
        "elevation": 2,
        "flexDirection": "row",
        "height": 44,
        "marginVertical": 10,
        "paddingHorizontal": 10,
        "width": "100%",
      }
    }
  >
    <Text />
    <TextInput
      placeholder="Tìm kiếm bãi đỗ"
      placeholderTextColor="#FF953A"
      style={
        Array [
          Object {
            "flex": 1,
            "fontSize": 20,
            "marginLeft": 10,
          },
          undefined,
        ]
      }
    />
  </View>
</ViewManagerAdapter_ExpoLinearGradient>
`;
